External/Plugins/HaXeContext/Context.cs

```diff
@@ -1381,10 +1399,60 @@ internal void OnFunctionCompletionResult(HaxeComplete hc, HaxeCompleteStatus sta
 
                 case HaxeCompleteStatus.TYPE:
                     hc.Expr.Position--;
-                    ASComplete.FunctionContextResolved(hc.Sci, hc.Expr, hc.Type, null, true);
+                    ASComplete.FunctionContextResolved(hc.Sci, hc.Expr, result.Type, null, true);
                     break;
             }
         }
+
+        public override bool HandleGotoDeclaration(ScintillaControl sci, ASExpr expression)
+        {
+            if (GetCurrentSDKVersion().IsOlderThan(new SemVer("3.2.0")))
+                return false;
+
+            var hc = new HaxeComplete(sci, expression, false, completionModeHandler, HaxeCompilerService.POSITION);
+            hc.GetPosition(OnPositionCompletionResult);
+            return true;
+        }
+
+        internal void OnPositionCompletionResult(HaxeComplete hc, HaxePositionCompleteResult result, HaxeCompleteStatus status)
+        {
+            switch (status)
+            {
+                case HaxeCompleteStatus.ERROR:
+                    TraceManager.AddAsync(hc.Errors, -3);
+                    break;
+
+                case HaxeCompleteStatus.POSITION:
+                    if (hc.Sci.InvokeRequired)
+                    {
+                        hc.Sci.BeginInvoke((MethodInvoker)delegate
+                        {
+                           HandlePositionCompletionResult(hc, result); 
+                        });
+                    }
+                    else HandlePositionCompletionResult(hc, result);
+                    break;
+            }
+        }
+
+        private void HandlePositionCompletionResult(HaxeComplete hc, HaxePositionCompleteResult result)
+        {
+            ASComplete.SaveLastLookupPosition(hc.Sci);
+
+            PluginBase.MainForm.OpenEditableDocument(result.Path, false);
+            ScintillaControl sci = PluginBase.MainForm.CurrentDocument.SciControl;
+            const string keywords = "(function|var|[,(])";
+
+            if (sci.InvokeRequired)
```

Comment:
It doesn't matter: we use `InvokeRequired` to join the UI thread, it can be done with any UI element you have a reference to.